doc/working_example.md

```diff
@@ -0,0 +1,825 @@
+code_fmt = gn.CodeFormat.TWOS_COMPLEMENT # integer data format
+fs = 3e6 # sample-rate of the data
+
+#
+# FFT configuration
+#
+navg = 1 # number of FFT averages
+nfft = int(npts/navg) # FFT-order
+qres = 12  # data resolution
+code_fmt = gn.CodeFormat.TWOS_COMPLEMENT # integer data format
+window = gn.Window.NO_WINDOW # window function to apply
+
+#
+# Genarate signal for analysis
+#
+awfi = gn.cos(npts, fs, ampl, freq, phase)
+awfq = gn.sin(npts, fs, ampl, freq, phase)
+qwfi = gn.quantize(awfi, fsr, qres, qnoise, code_fmt)
+qwfq = gn.quantize(awfq, fsr, qres, qnoise, code_fmt)
+
+#
+# Compute FFT
+#
+fft_cplx = gn.fft(qwfi, qwfq, qres, navg, nfft, window, code_fmt)
+
+#
+# Fourier analysis configuration
+#
+object_key = "fa" # object-key
+signal_tone_key = 'A' # signal-key
+ssb_fund = 0 # number of single-side bins
+gn.fa_create(object_key)
+gn.fa_fsample(object_key, fs)
+gn.fa_max_tone(object_key, signal_tone_key, gn.FaCompTag.SIGNAL, ssb_fund)
+```
+</details>
+
+##### Create ``fourier_analysis`` Object
+The first step in configuring Genalyzer is to call the ``fa_create()`` function with an _object\_key_. This key is used for all configuring Genalyzer further, and for computing and retrieving the metrics.
+```{admonition} Aside
+:class: note
+
+Under the hood, Genalyzer adds a key-value pair to a  ``static`` ``map`` container to manage the metrics to be computed. The key is the string argument passed through ``fa_create()``, and the mapped value is a shared-pointer to an instance of ``fourier_analysis`` class. This key is then used to further configure Genalyzer, and to compute and retrieve the metrics through ``fourier_analysis`` class. The intent behind using a ``map`` container is to be easily able to associate multiple keys to different snapshots of the data being analyzed and to have the metrics for each of those snapshots available. 
+```
+
+##### Set Sample-Rate
+Set the sample-rate of the data vector using ``fa_fsample``. As shown on this [page](#genalyzer.advanced.advanced.fa_fsample), we associate this configuration step with an _object\_key_.
+<br/><br/>
+
+##### Identify Signal Tone
+The next step is to identify the signal tone and label it. This is done by calling either [``fa_fixed_tone()``](#genalyzer.advanced.advanced.fa_fixed_tone) or [``fa_max_tone()``](#genalyzer.advanced.advanced.fa_max_tone) functions. In both cases, the first two arguments are _keys_. The first _key_ is the _object\_key_ mentioned previously, and the second _key_ is a string that is to be associated with the signal component whose magnitude is measured and used subsequently when computing performance metrics. In both cases, the third argument is a tag which maps to the component type. To identify a component as the signal tone, we use ``GnFACompTagSignal``. The other tags available are shown [here](#genalyzer.advanced.advanced.FaCompTag). As the name indicates, ``fa_max_tone()`` interprets the tone with the highest magnitude as the signal tone.
```

Comment:
>The next step is to identify the signal tone and label it. This is done by calling either [``fa_fixed_tone()``](#genalyzer.advanced.advanced.fa_fixed_tone) or [``fa_max_tone()``](#genalyzer.advanced.advanced.fa_max_tone) functions. In both cases, the first two arguments are _keys_. The first _key_ is the _object\_key_ mentioned previously, and the second _key_ is a string that is to be associated with the signal component whose magnitude is measured and used subsequently when computing performance metrics. In both cases, the third argument is a tag which maps to the component type. To identify a component as the signal tone, we use ``GnFACompTagSignal``. The other tags available are shown [here](#genalyzer.advanced.advanced.FaCompTag). As the name indicates, ``fa_max_tone()`` interprets the tone with the highest magnitude as the signal to

The descriptions of keys and tags here is a bit confusing. Is there a plain way to describe what they are for?